### PR TITLE
Input parameter check & Error message modification

### PR DESCRIPTION
**Pull request by amberbhardwaj**
Added null pointer checking
Added input parameter checking
Added return values in non-void functions
Added proper error message for debugging purpose

Signed-off-by: amberbhardwaj <amber.bhardwaj10@gmail.com>

### CODE REVIEW

**Comment by amberbhardwaj** [comment]
Hello @clipka ,

This is the latest patch with sign-off. I closed the previous patch so this patch is now ready to merge with the same changes.

Thanks :)

**Comment by gunnarx** [comment]
@amberbhardwaj You can update a PR instead of creating a new  a one, by pushing new changes to the source branch. (If the changes are a rewrite you use forced push)

**Comment by amberbhardwaj** [comment]
@gunnarx yeah but with my older PR, i did a mistake. 
I pushed the changes for dlt-kpi.c and dlt-kpi-process.c but when I checked after pushing, it was showing dlt-system.conf as a new change. However, my patch for dlt-system.conf already merged long back so that is the only reason I removed the older patch and pushed the fresh patch with kpi file changes.

File: src/kpi/dlt-kpi-process.c [hunk 1]
```diff
@@ -275,8 +288,8 @@ DltReturnValue dlt_kpi_print_process(DltKpiProcess *process)
 {
     if(process == NULL)
     {
-        fprintf(stderr, "Error: Nullpointer parameter\n");
-        return DLT_RETURN_ERROR;
+        fprintf(stderr, "dlt_kpi_print_process: Nullpointer parameter\n");
```
Comment (clipka) [inline src/kpi/dlt-kpi-process.c]: I'd prefer:
`fprintf(stderr, "%s: Invalid parameter (NULL)\n", __func__); `

**Comment by amberbhardwaj** [comment]
@clipka 
Ok, i will modify this. 

One more thing is I am not able to push directly to genivi dlt.[Getting Permission Denied] 

So I opted an alternate way. I created my own local repo and fork genivi dlt-daemon then clone my repo in my system. Now I am making changes in my system's local copy then I push those change in my own forked branch (dlt-daemon) and from that branch (using git web) I am comparing master branch(Main Genivi DLT). Then I am creating a new pull request after comparing.

So please suggest me any other simple solution for pushing the code changes.

Thanks

**Comment by clipka** [comment]
You cannot push directly into this repo. The way you are doing is basically correct,

**Comment by amberbhardwaj** [comment]
Hello @clipka,
I have pushed new patch with some additional modification. Please review it.
Commit ID for change is bb32d960bdd03ad0b7db9ad2b93a60cc5bf9a183

Thanks,
Amber

**Pull request merged**